mce-hal.c

```diff
@@ -40,6 +40,21 @@ product_id_t get_product_id(void)
 	if (product_id != PRODUCT_UNSET)
 		goto EXIT;
 
+	if (mce_read_string_from_file(DEVICETREE_MODEL_PATH, &tmp)) {
+		if (strstr(tmp, DEVICETREE_MODEL_XT894) != NULL) {
+			product_id = PRODUCT_XT894;
+		}
+		g_free(tmp);
+	}
+	/*  We should probably transition all other devices to device tree model
+		names, but I don't have all the devices here, so let's also check
+		/proc/component_version
+	*/
+
+	if (product_id != PRODUCT_UNSET)
+		goto EXIT;
```

Comment:
It's better to get rid of such goto's - they can be replaced with a single "return" statement.

Reply:
In this case I add support for a new product matching, and the intention is not to rewrite how the function works, so this can be done in a different PR/commit.